File: forest/village-guard.lua
function patrol(x, y)
    self:moveXY(x, y)
    local e = self:findNearest(self:findEnemies())
    if e then
        while e.health > 0 do
            self:attack(e)
        end
    end
end

loop
    patrol(35, 34)
    patrol(60, 31)
end

File: forest/village-guard.clj
(defn patrol [x y]
  (.moveXY this x y)
  (def e (.findNearest this (.findEnemies this)))
  (if e
    (while (> (.health e) 0)
       (.attack this e))))

(while true
  (patrol 35 34)
  (patrol 60 31))
